src/main/java/com/denys/toukdemo/controller/ReservationController.java

```diff
@@ -0,0 +1,80 @@
+package com.denys.toukdemo.controller;
+
+import com.denys.toukdemo.dto.Reservation;
+import com.denys.toukdemo.dto.ReservationUnit;
+import com.denys.toukdemo.dto.Room;
+import com.denys.toukdemo.dto.User;
+import com.denys.toukdemo.helper.RoomHelper;
+import com.denys.toukdemo.service.MailService;
+import com.denys.toukdemo.service.ReservationService;
+import com.denys.toukdemo.validation.UserValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@RestController
+public class ReservationController {
+
+    private static final Integer VOUCHER_DISCOUNT_GRADE = 2;
+
+    @Autowired
+    ReservationService reservationService;
+
+    @Autowired
+    MailService mailService;
+
+    @RequestMapping(value = "/particularscreening", method = RequestMethod.GET)
+    public Room getParticularScreeningBySeans(@RequestParam String seansId){
+        return reservationService.getRoomWithReservedPlacesBySeansId(seansId);
+    }
+
+    @RequestMapping(value = "/reserveseats", method = RequestMethod.POST)
+    public String reserveSeats(@RequestBody Reservation reservation){
+        if(!UserValidator.isValid(reservation.getOwner()))
+            return "Invalid user data";
+
+        Room room = getParticularScreeningBySeans(reservation.getSeans().getId().toString());
```

Comment:
czy ta logika powinna być w kontrolerze?

Reply:
> czy ta logika powinna być w kontrolerze?

brak doświadczenia z aplikacjami takiego typu
można to wynieść w warstwę service